fuzzing/replay/status_util.cc

```diff
@@ -26,19 +26,23 @@ namespace {
 
 constexpr size_t kMaxErrorStringSize = 128;
 
-std::string StrError(int errno_value) {
-  char error_str_buf[kMaxErrorStringSize];
-#if (_POSIX_C_SOURCE >= 200112L) && !_GNU_SOURCE
-  const int result =
-      strerror_r(errno_value, error_str_buf, sizeof(error_str_buf));
+std::string WrapError(char* result, int errno_value, char* error_str_buf) {
+  return result;
+}
+
+std::string WrapError(int result, int errno_value, char* error_str_buf) {
   if (result) {
     return absl::StrCat("Unknown error ", errno_value);
   } else {
-    return error_str_buf;
+    return WrapError(error_str_buf, errno_value, error_str_buf);
   }
-#else
-  return strerror_r(errno_value, error_str_buf, sizeof(error_str_buf));
-#endif
+}
```

Comment:
Since we're changing this anyway, perhaps let's adopt the logic used in ABSL for consistency? It's also known to work across more platforms. https://github.com/abseil/abseil-cpp/blob/3b4a16abad2c2ddc494371cc39a2946e36d35d11/absl/base/internal/strerror.cc#L32-L51

Ideally, we could have directly included that code, alas its build target has restricted visibility.

Reply:
Done.
Agree. Perhaps we should file an issue to make this a part of public API?